joomla/joomla-cms
# [4.0] Change ModuleChromes to JLayouts

Inspired on some ideas from Michael in https://github.com/joomla/joomla-cms/issues/20864 and some recent discussions about the Module Class behavior.
For those who aren't aware: The Module Chrome is the part around the actual module output. In the module settings, it is called "Module Style".

### Summary of Changes
This PR moves the module chromes (the code in that funny /templates/yourtemplate/html/modules.php file) to JLayouts.
The ones from the system templates are in the global /layouts/chromes/ folder.
The ones from the core templates are in the respective templates html/layouts/chromes/ folder.

#### Backward compatibility related
Support for chromes defined in modules.php is removed. Tenplates who wish to support both J3 and J4 can do so by adding the chromes as JLayouts and still keep the modules.php. In J3, the legacy one is used and in J4 the JLayouts are used.

The protected function `Joomla\CMS\Form\Field\ChromestyleField::getSystemTemplate()` is removed since it no longer is used.

Existing settings in modules for the module style keep working as the value stored in J3 and J4 is exactly the same.

### Testing Instructions
Set different module styles and test if the module is rendered with the expected chrome.
Test if you can add new layouts or override the existing chromes.

### Expected result
* Works from the JLayouts
* Custom chromes are possible
* Chromes are overrideable

### Actual result
* Works, but uses modules.php
* No customisation possible


### Documentation Changes Required
If module chromes are documented somewhere, that needs to be adjusted.
